File: app/src/main/java/com/adsamcik/signalcollector/Receivers/BatteryStateReceiver.java
package com.adsamcik.signalcollector.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.adsamcik.signalcollector.services.TrackerService;
import com.adsamcik.signalcollector.Preferences;

public class BatteryStateReceiver extends BroadcastReceiver {
	@Override
	public void onReceive(Context context, Intent intent) {
		switch(intent.getAction()) {
			case Intent.ACTION_BATTERY_LOW:
				Preferences.stopTillRecharge(context);
				if(TrackerService.service != null)
					context.stopService(TrackerService.service);
				break;
		}
	}
}

File: app/src/main/java/com/adsamcik/signalcollector/receivers/BatteryStateReceiver.kt
package com.adsamcik.signalcollector.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.adsamcik.signalcollector.services.TrackerService
import com.adsamcik.signalcollector.utility.Preferences

class BatteryStateReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        if (action != null) {
            when (action) {
                Intent.ACTION_BATTERY_LOW -> {
                    Preferences.stopTillRecharge(context)
                    if (TrackerService.isRunning)
                        context.stopService(Intent(context, TrackerService::class.java))
                }
            }
        }
    }
}
